examples/helloworld/lib/server.ex

```diff
@@ -3,6 +3,17 @@ defmodule Helloworld.Greeter.Server do
 
   @spec say_hello(Helloworld.HelloRequest.t(), GRPC.Server.Stream.t()) ::
           Helloworld.HelloReply.t()
+  def say_hello(%{name: "quota_failure:" <> name}, stream) do
+    bin = Google.Rpc.QuotaFailure.new(
+      violations: [%{subject: "name:#{name}", description: "Limit one greeting per person"}]
+    )
+    |> Protobuf.encode()
+    details = Google.Protobuf.Any.new(type_url: "type.googleapis.com/google.rpc.QuotaFailure", value: bin)
+    |> Protobuf.encode()
+    stream = GRPC.Server.set_headers(stream, %{"grpc-status-details-bin" => details})
```

Comment:
No need, since what it does in the end is sending new headers to separate process which owns the data: https://github.com/surgeventures/grpc/blob/762189ba87a6b002a0decf2476e694ecae063952/lib/grpc/adapter/cowboy/handler.ex#L130